crates/ruma-state-res/src/event_auth.rs

```diff
@@ -343,10 +345,19 @@ pub fn auth_check<E: Event>(
         }
     } else {
         // If no power level event found the creator gets 100 everyone else gets 0
-        from_json_str::<RoomCreateEventContent>(room_create_event.content().get())
-            .ok()
-            .and_then(|create| (create.creator == *sender).then(|| int!(100)))
-            .unwrap_or_default()
+        let is_creator = if room_version.use_room_create_sender {
+            room_create_event.sender() == sender
+        } else {
+            #[allow(deprecated)]
+            from_json_str::<RoomCreateEventContent>(room_create_event.content().get())
+                .is_ok_and(|create| create.creator.unwrap() == *sender)
```

Comment:
Is it ok to bumpt the MSRV for that?

Reply:
We weren't using this before? Also, I don't see an MSRV bump.

Reply:
No we weren't, our MSRV is 1.65. I didn't do the MSRV bump because I was waiting for the answer, since it's mostly useful for CI.

Reply:
Yeah, seems fine to bump MSRV. Maybe in a separate PR, where existing `map_or`s can be converted?